pkg/server/cstorvolumeconfig/backup_endpoint.go

```diff
@@ -130,32 +142,41 @@ func (bOps *backupAPIOps) create() (interface{}, error) {
 	// find healthy CVR which will helps to create backup CR
 	cvr, err := findHealthyCVR(bOps.clientset, backup.Spec.VolumeName, bOps.namespace)
 	if err != nil {
-		return nil, CodedError(400, fmt.Sprintf("Failed to find healthy replica"))
+		return nil, CodedError(400, fmt.Sprintf("Failed to find healthy replica for volume %s", backup.Spec.VolumeName))
 	}
 
+	poolName := cvr.ObjectMeta.Labels[cstortypes.CStorPoolInstanceNameLabelKey]
 	backup.ObjectMeta.Labels = map[string]string{
 		cstortypes.CStorPoolInstanceUIDLabelKey:  cvr.ObjectMeta.Labels[cstortypes.CStorPoolInstanceUIDLabelKey],
-		cstortypes.CStorPoolInstanceNameLabelKey: cvr.ObjectMeta.Labels[cstortypes.CStorPoolInstanceNameLabelKey],
+		cstortypes.CStorPoolInstanceNameLabelKey: poolName,
 		cstortypes.PersistentVolumeLabelKey:      cvr.ObjectMeta.Labels[cstortypes.PersistentVolumeLabelKey],
 		"openebs.io/backup":                      backup.Spec.BackupName,
 	}
 
-	// Find last backup snapshot name
-	lastsnap, err := bOps.getLastBackupSnap(backup)
+	poolVersion, err := getPoolVersion(poolName, bOps.namespace, bOps.clientset)
 	if err != nil {
-		return nil, CodedError(400, fmt.Sprintf("Failed create lastbackup"))
+		return nil, CodedError(400, fmt.Sprintf("failed to get %s pool version error: %v", poolName, err))
 	}
 
+	backupInterface, err := bOps.getBackupInterfaceFromPoolVersion(poolVersion, backup)
+	if err != nil {
+		return nil, CodedError(400, fmt.Sprintf("failed to get backupInterface error: %v", err))
+	}
+
+	lastSnapName, err := backupInterface.getOrCreateLastBackupSnap()
+	if err != nil {
+		return nil, CodedError(400, fmt.Sprintf("Failed get or create lastCompleted backup error: %v", err.Error()))
+	}
 	// Initialize backup status as pending
-	backup.Status = openebsapis.BKPCStorStatusPending
-	backup.Spec.PrevSnapName = lastsnap
+	backupInterface.setBackupStatus(string(openebsapis.BKPCStorStatusPending))
+	backupInterface.setLastSnapshotName(lastSnapName)
 
+	// NOTE: For logging we are using v1 backup resource itself
```

Comment:
Not able to understand this logging part

Reply:
Updated the comment... Basically, when processing the backup request we are logging by using the v1 version of backup irrespective of the request version either it can v1/v1alpha.